File: 0412-fizz-buzz/0412-fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    let result = [];
    for(let i = 1; i <= n; i++) {
        switch(true) {
            case i % 3 === 0 && i % 5 === 0:
                result.push("FizzBuzz");
                break;
            case i % 3 === 0:
                result.push("Fizz");
                break;
            case i % 5 === 0:
                result.push("Buzz");
                break;
            default:
                result.push(i.toString());     
        }

    }
    return result;
};

File: 0412-fizz-buzz/0412-fizz-buzz.go
func fizzBuzz(n int) []string {
	result := make([]string, 0, n)
	for i := 1; i <= n; i++ {
		switch {
		case i%3 == 0 && i%5 == 0:
			result = append(result, "FizzBuzz")
		case i%3 == 0:
			result = append(result, "Fizz")
		case i%5 == 0:
			result = append(result, "Buzz")
		default:
			result = append(result, fmt.Sprintf("%d", i))
		}
	}
	return result
}
